index.js

```diff
@@ -70,7 +70,9 @@ class ExecutorQueue extends Executor {
             this.redis[funcName](...args), breaker);
         this.requestRetryStrategy = (err, response) =>
             !!err || (response.statusCode !== 201 && response.statusCode !== 200);
-
+        this.requestRetryStrategyPostEvent = (err, response) =>
+            !!err || (response.statusCode !== 201 && response.statusCode !== 200
+            && response.statusCode !== 404); // postEvent can return 404 if no job to start
```

Comment:
is 404 the only status we need to handle ? shouldn't we retry only if error status is 5xx ?

Reply:
only `404` is a known behavior we defined for `no job to start`. For this case we know for sure it's not an issue. For other status code i think we should still retry. It will preserve the current behavior.